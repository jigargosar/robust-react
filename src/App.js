import {
  AppBar,
  CssBaseline,
  List,
  MenuItem,
  Paper,
  TextField,
  Toolbar,
  Typography,
  withStyles,
} from 'material-ui'
import {div, h} from './hyper-script'
import S from './sanctuary'

const NumberList = () =>
  h(
    List,
    S.pipe(
      [S.range(0), S.reverse, S.map(id => h(MenuItem, {key: id}, id + 1))],
      20,
    ),
  )

const NumberListLayout = withStyles(theme => ({
  root: {
    overflow: 'scroll',
    flex: 1,
  },
  container: {
    display: 'flex',
    flexDirection: 'column-reverse',
  },
  paper: {margin: theme.spacing.unit * 2},
}))(({classes}) =>
  div({className: classes.root}, [
    div({className: classes.container}, [
      h(Paper, {className: classes.paper}, [h(NumberList)]),
    ]),
  ]),
)

const Header = withStyles(({spacing: {unit}}) => ({
  flex: {
    flex: 1,
  },
  textFieldRoot: {
    // backgroundColor: 'rgba(0,0,0,0.1)',
    backgroundColor: 'rgba(255,255,255,0.1)',
    color: 'inherit',
  },
  textFieldInput: {
    margin: `0 ${unit}px`,
  },
}))(({classes}) =>
  h(AppBar, {position: 'static'}, [
    h(Toolbar, [
      h(
        Typography,
        {
          className: classes.flex,
          variant: 'title',
          color: 'inherit',
        },
        'Robust React App',
      ),
      h(TextField, {
        type: 'search',
        placeholder: 'Search',
        InputProps: {
          disableUnderline: true,
          classes: {
            root: classes.textFieldRoot,
            input: classes.textFieldInput,
          },
        },
      }),
    ]),
  ]),
)
const Footer = withStyles({})(() =>
  h(AppBar, {position: 'static', component: 'footer'}, [
    h(Toolbar, [h(Typography, {variant: 'title', color: 'inherit'}, 'Footer')]),
  ]),
)
const Layout = withStyles({
  root: {
    height: '100vh',
    width: '100vw',
    overflow: 'hidden',
    display: 'flex',
    flexDirection: 'column',
  },
})(({children, classes}) => div({className: classes.root}, [children]))

const App = () =>
  h(CssBaseline, [h(Layout, [h(Header), h(NumberListLayout), h(Footer)])])

export default App
