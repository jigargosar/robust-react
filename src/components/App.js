import {
  AppBar,
  CssBaseline,
  Paper,
  Toolbar,
  Typography,
  withStyles,
} from 'material-ui'
import * as MR from 'mobx-react'
import * as R from 'ramda'
import {div, h} from '../hyper-script'
import Header from './Header'
import ModelList from './model/List'

const ModelListContainer = withStyles(theme => ({
  root: {
    overflow: 'scroll',
    flex: 1,
  },
  paper: {
    margin: theme.spacing.unit * 2,
  },
}))(({classes}) =>
  div({className: classes.root}, [
    h(Paper, {className: classes.paper}, [h(ModelList)]),
  ]),
)

const Footer = withStyles({})(() =>
  h(AppBar, {position: 'static', component: 'footer'}, [
    h(Toolbar, [h(Typography, {variant: 'title', color: 'inherit'}, 'Footer')]),
  ]),
)
const Layout = withStyles({
  root: {
    height: '100vh',
    width: '100vw',
    overflow: 'hidden',
    display: 'flex',
    flexDirection: 'column',
  },
})(({children, classes}) => div({className: classes.root}, [children]))

const injectObserve = R.compose(
  MR.inject((stores, props) => ({
    ...stores,
    ...props,
  })),
  MR.observer,
)

const App = () => {
  return h(CssBaseline, [
    h(Layout, [h(Header), h(ModelListContainer), h(Footer)]),
  ])
}

export default injectObserve(App)
