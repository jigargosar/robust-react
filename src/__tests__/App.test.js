import React from 'react'
import {render} from 'react-testing-library'
import App from '../App'
import JssProvider from 'react-jss/lib/JssProvider'

const generateClassName = (rule, styleSheet) =>
  `${styleSheet.options.classNamePrefix}-${rule.key}`

it('renders without crashing', () => {
  const {getByText, getByTestId, container} = render(
    <JssProvider generateClassName={generateClassName}>
      <App/>
    </JssProvider>
  )
  expect(container.firstChild).toMatchSnapshot()
})
