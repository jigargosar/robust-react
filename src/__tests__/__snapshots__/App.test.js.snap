// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  class="Component-root-1"
>
  <header
    class="MuiPaper-root-10 MuiPaper-elevation4-16 MuiAppBar-root-2 MuiAppBar-positionStatic-6 MuiAppBar-colorPrimary-8"
  >
    <div
      class="MuiToolbar-root-37 MuiToolbar-gutters-38"
    >
      <h2
        class="MuiTypography-root-39 MuiTypography-title-45 MuiTypography-colorInherit-58"
      >
        Robust React App
      </h2>
    </div>
  </header>
  <div
    class="Component-root-63"
  >
    <div
      class="MuiPaper-root-10 MuiPaper-elevation2-14 MuiPaper-rounded-11 Component-paper-64"
    >
      <ul
        class="MuiList-root-66 MuiList-padding-67 Component-list-65"
      >
        <li
          class="MuiButtonBase-root-82 MuiListItem-root-72 MuiListItem-default-75 MuiListItem-gutters-79 MuiListItem-button-80 MuiMenuItem-root-70"
          role="menuitem"
          tabindex="-1"
        >
          0
          <span
            class="MuiTouchRipple-root-85"
          />
        </li>
        <li
          class="MuiButtonBase-root-82 MuiListItem-root-72 MuiListItem-default-75 MuiListItem-gutters-79 MuiListItem-button-80 MuiMenuItem-root-70"
          role="menuitem"
          tabindex="-1"
        >
          1
          <span
            class="MuiTouchRipple-root-85"
          />
        </li>
        <li
          class="MuiButtonBase-root-82 MuiListItem-root-72 MuiListItem-default-75 MuiListItem-gutters-79 MuiListItem-button-80 MuiMenuItem-root-70"
          role="menuitem"
          tabindex="-1"
        >
          2
          <span
            class="MuiTouchRipple-root-85"
          />
        </li>
        <li
          class="MuiButtonBase-root-82 MuiListItem-root-72 MuiListItem-default-75 MuiListItem-gutters-79 MuiListItem-button-80 MuiMenuItem-root-70"
          role="menuitem"
          tabindex="-1"
        >
          3
          <span
            class="MuiTouchRipple-root-85"
          />
        </li>
        <li
          class="MuiButtonBase-root-82 MuiListItem-root-72 MuiListItem-default-75 MuiListItem-gutters-79 MuiListItem-button-80 MuiMenuItem-root-70"
          role="menuitem"
          tabindex="-1"
        >
          4
          <span
            class="MuiTouchRipple-root-85"
          />
        </li>
        <li
          class="MuiButtonBase-root-82 MuiListItem-root-72 MuiListItem-default-75 MuiListItem-gutters-79 MuiListItem-button-80 MuiMenuItem-root-70"
          role="menuitem"
          tabindex="-1"
        >
          5
          <span
            class="MuiTouchRipple-root-85"
          />
        </li>
        <li
          class="MuiButtonBase-root-82 MuiListItem-root-72 MuiListItem-default-75 MuiListItem-gutters-79 MuiListItem-button-80 MuiMenuItem-root-70"
          role="menuitem"
          tabindex="-1"
        >
          6
          <span
            class="MuiTouchRipple-root-85"
          />
        </li>
        <li
          class="MuiButtonBase-root-82 MuiListItem-root-72 MuiListItem-default-75 MuiListItem-gutters-79 MuiListItem-button-80 MuiMenuItem-root-70"
          role="menuitem"
          tabindex="-1"
        >
          7
          <span
            class="MuiTouchRipple-root-85"
          />
        </li>
        <li
          class="MuiButtonBase-root-82 MuiListItem-root-72 MuiListItem-default-75 MuiListItem-gutters-79 MuiListItem-button-80 MuiMenuItem-root-70"
          role="menuitem"
          tabindex="-1"
        >
          8
          <span
            class="MuiTouchRipple-root-85"
          />
        </li>
        <li
          class="MuiButtonBase-root-82 MuiListItem-root-72 MuiListItem-default-75 MuiListItem-gutters-79 MuiListItem-button-80 MuiMenuItem-root-70"
          role="menuitem"
          tabindex="-1"
        >
          9
          <span
            class="MuiTouchRipple-root-85"
          />
        </li>
      </ul>
    </div>
  </div>
</div>
`;
